src/http_protocol.cpp

```diff
@@ -731,4 +754,42 @@ std::string http_response_t::generic(unsigned code,
   return response;
 }
 
-} // namespace prime_server
+// namespace prime_server
+}
+
+// make a short circuiter that will respond to health checks and OPTIONS request
+// Ref: https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/OPTIONS
+prime_server::shortcircuiter_t<prime_server::http_request_t> make_shortcircuiter(const uint8_t& verb_mask = std::numeric_limits<uint8_t>::max(),
+                                                                                      const std::string& health_check_path = "/health_check") {
```

Comment:
the defaults go into the declaration not the definition move them to the header and put this function in the namespace

Reply:
Done